packages/fiori-annotation-api/src/change-converter.ts

```diff
@@ -582,23 +590,26 @@ export class ChangeConverter {
                 this.annotationFileChanges.push(internalChange);
             } else if (node.attributes[type]) {
                 // attribute notation
-                const internalChange: ReplaceAttribute = {
+                this.annotationFileChanges.push({
                     type: REPLACE_ATTRIBUTE,
                     uri: file.uri,
                     pointer: pointer + `/attributes/${type}`,
                     newAttributeName: content.value.type,
                     newAttributeValue: newValue
-                };
-                this.annotationFileChanges.push(internalChange);
+                });
             } else if (node.name === valueType) {
                 // element notation
+                let childContent: ElementChild[] = [];
+                if (content.value.type !== Edm.Null) {
+                    childContent.push(createTextNode(newValue));
+                }
                 const internalChange: ReplaceElement = {
                     type: REPLACE_ELEMENT,
                     uri: file.uri,
                     pointer: pointer,
                     newElement: createElementNode({
                         name: content.value.type,
-                        content: [createTextNode(newValue)]
+                        content: [...childContent]
```

Comment:
We do not need to make a copy of the array, we can just use it here.

Reply:
adapted